azurelinuxagent/common/utils/restutil.py

```diff
@@ -310,7 +310,7 @@ def redact_sas_tokens_in_urls(url):
 
 
 def _http_request(method, host, rel_uri, port=None, data=None, secure=False,
-                  headers=None, proxy_host=None, proxy_port=None, redact_data=False):
+                  headers=None, proxy_host=None, proxy_port=None, redact_data=False, timeout=10):
```

Comment:
We defining the default timeout in multiple places. We should probably do that on the higher level functions (e.g. http_put) and make a mandatory parameter in the other methods (http_request, _http_request)

Reply:
That would force us to change every http_* API call (GET, POST, etc). I'm okay to do that too.

Reply:
yes, I think that is a better API interface (which requires more code changes, but hey...) :)

Reply:
(Unless you think otherwise)

Reply:
Sometimes better things come with price. I'm gonna pay for it if things are clean :)